src/plugins/saved_objects_management/public/lib/filter_query.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+export function filterQuery(allowedVals: string[], requestedVals: string[]): string[] {
```

Comment:
Shouldn't `requestedVals` be optional? I ask because the condition below allows for that.

Reply:
Added `?` to the arg to indicate optionality.